src/charm.py

```diff
@@ -976,70 +1056,106 @@ def _core_reconciliation(self):
         if self._current_wp_config() is not None:
             self._start_server()
 
-    def _wp_theme_list(self):
+    def _check_addon_type(self,addon_type):
+        """Check if addon_type is one of the accepted addon types (theme/plugin).
+
+        Raise a ValueException if not.
+        """
+        if addon_type not in ("theme", "plugin"):
+            raise ValueError(f"Addon type unknown {repr(addon_type)}, accept: (theme, plugin)")
+
+    def _wp_addon_list(self, addon_type):
```

Comment:
Is it addon or themes in this case? I believe it could be either a theme or plugin, should the docstring be updated?

Reply:
Fixed, thanks!